conans/client/toolchain/msbuild.py

```diff
@@ -32,14 +34,20 @@ def command(self, sln):
                                 vcvars_ver=None)
         cmd = ('%s && msbuild "%s" /p:Configuration=%s /p:Platform=%s '
                % (vcvars, sln, self.build_type, self.platform))
+
+        if self.compiler == 'intel':
+            cvars = intel_compilervars_command(self._conanfile)
+            cmd = '%s && %s' % (cvars, cmd)
+            cmd += '/p:PlatformToolset="%s"' % msvs_toolset(self._conanfile)
```

Comment:
Is this line really necessary if the toolset is defined in the toolchain file? It was one of the goals of the toolchain, to simplify the command line invocation by capturing some of it in the toolchain files

Reply:
for me, it doesn't work without this line. probably, it's too late in toolchain file to override this

Reply:
It works well for other toolsets, that behavior is tested, in the MSBuildToolchain tests.

Reply:
Maybe it was before I moved the import of the toolchain to the correct place?

Reply:
yes, checked one more time and it works! sorry for the confusion, I'll create a follow up PR.

Reply:
see #8078